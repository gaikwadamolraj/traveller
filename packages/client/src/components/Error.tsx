import { Alert, AlertIcon } from '@chakra-ui/react'
import type { IErrorProps } from '../dataStructure'

export const Error: React.FC<IErrorProps> = ({ error }) => {
  if (error?.length) {
    return (
      <Alert status="error">
        <AlertIcon />
        {error}
      </Alert>
    )
  }
  return <></>
}
