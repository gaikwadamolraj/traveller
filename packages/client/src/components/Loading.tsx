import { Spinner } from '@chakra-ui/react'
import type { IEmptyProps } from '../dataStructure'

export const Loading: React.FC<IEmptyProps> = () => (
  <Spinner thickness="4px" speed="0.65s" emptyColor="gray.200" color="blue.500" size="xl" />
)
