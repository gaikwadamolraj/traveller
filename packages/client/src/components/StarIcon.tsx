import { StarIcon } from '@chakra-ui/icons'
import type { StarIconProps } from '../dataStructure'

export const StarIconWithAction: React.FC<StarIconProps> = ({ page, city, color, handleFunc }) => {
  if (typeof handleFunc === 'function') {
    return <StarIcon onClick={() => handleFunc(city, page)} marginLeft={'20px'} color={color} />
  }

  return <StarIcon marginLeft={'20px'} color={color} />
}
