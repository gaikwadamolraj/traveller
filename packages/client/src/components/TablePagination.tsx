import * as React from 'react'
import { Table } from 'react-chakra-pagination'
import { Box, useColorMode } from '@chakra-ui/react'
import { FiDatabase } from 'react-icons/fi'
import type { ICity, TableProps } from '../dataStructure'
import { CONSTANTS } from '../dataStructure'
import { StarIconWithAction } from './StarIcon'

const tableColumns = [
  {
    Header: 'Name',
    accessor: 'name' as const,
  },
  {
    Header: 'Country',
    accessor: 'country' as const,
  },
  {
    Header: 'Wish List',
    accessor: 'wishAction' as const,
  },
  {
    Header: 'Visited',
    accessor: 'visitAction' as const,
  },
]

export const TablePagination: React.FC<TableProps> = ({
  pageNumber = CONSTANTS.DEFAULT_PAGE,
  cities,
  handleWishlist,
  handleVisited,
}: TableProps) => {
  const [page, setPage] = React.useState(pageNumber)
  const { colorMode } = useColorMode()

  const tableData = cities?.map((city: ICity) => ({
    name: city.name,
    country: city.country,
    wishAction: (
      <StarIconWithAction
        page={page}
        city={city}
        color={city.wishlist ? 'green' : 'gray'}
        handleFunc={handleWishlist}
      />
    ),
    visitAction: (
      <StarIconWithAction page={page} city={city} color={city.visited ? 'green' : 'gray'} handleFunc={handleVisited} />
    ),
  }))

  return (
    <Box>
      <Box>
        <Table
          key={'paginationTable'}
          colorScheme={colorMode === 'light' ? 'blue' : 'yellow'}
          emptyData={{
            icon: FiDatabase,
            text: 'No city found',
          }}
          totalRegisters={cities?.length}
          page={page}
          onPageChange={(page: number) => setPage(page)}
          columns={tableColumns}
          data={tableData}
        />
      </Box>
    </Box>
  )
}
