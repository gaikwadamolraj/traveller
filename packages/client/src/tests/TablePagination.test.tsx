import {TablePagination} from '../components/TablePagination'
import { render } from '../test-utils'


describe('<TablePagination /> component', () => {
  it('TablePagination component snapshot', () => {
      const cities = [{
        id: 0,
        name: 'Moscow',
        country: 'Russia',
        visited: false,
        wishlist: false,
      }]
    const { container } = render(<TablePagination cities={cities}/>)
    expect(container).toMatchSnapshot()
  })
})
