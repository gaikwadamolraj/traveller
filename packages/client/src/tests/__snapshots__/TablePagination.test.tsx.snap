// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TablePagination /> component TablePagination component snapshot 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-0"
    >
      <div
        class="css-1gpy5cb"
      >
        <table
          class="chakra-table css-5605sr"
          role="table"
        >
          <thead
            class="css-0"
          >
            <tr
              class="css-0"
              role="row"
            >
              <th
                class="css-1zts0j"
                colspan="1"
                role="columnheader"
              >
                Name
              </th>
              <th
                class="css-1zts0j"
                colspan="1"
                role="columnheader"
              >
                Country
              </th>
              <th
                class="css-1zts0j"
                colspan="1"
                role="columnheader"
              >
                Wish List
              </th>
              <th
                class="css-1zts0j"
                colspan="1"
                role="columnheader"
              >
                Visited
              </th>
            </tr>
          </thead>
          <tbody
            class="css-0"
            role="rowgroup"
          >
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-xumdn4"
                role="cell"
              >
                Moscow
              </td>
              <td
                class="css-xumdn4"
                role="cell"
              >
                Russia
              </td>
              <td
                class="css-xumdn4"
                role="cell"
              >
                <svg
                  class="chakra-icon css-17jvsh1"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                    fill="currentColor"
                  />
                </svg>
              </td>
              <td
                class="css-xumdn4"
                role="cell"
              >
                <svg
                  class="chakra-icon css-17jvsh1"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                    fill="currentColor"
                  />
                </svg>
              </td>
            </tr>
          </tbody>
        </table>
        <div
          class="chakra-stack css-10okdy6"
        >
          <div
            class="chakra-stack css-nd8846"
          >
            <button
              class="chakra-button css-7mygub"
              disabled=""
              type="button"
            >
              1
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
